### PR TITLE
Prevent unbinding of errorHandler.report

### LINKED ISSUE #40: Uncaught TypeError: Cannot read property 'reporter' of undefined in error-handler.js

### PR DESCRIPTION
**Pull request by lolmaus**
fixes #40

shame

### CODE REVIEW

**Comment by lolmaus** [comment]
That false sense of security clumsy tests give you...

**Comment by lolmaus** [comment]
Oops, lemme fix the tests.

**Comment by lolmaus** [comment]
Ah dammit, I've messed up the PR. :( I'll fix it.

**Comment by lolmaus** [comment]
All fixed. Ready for review.

**Comment by jherdman** [comment]
Would you mind squashing this down?

**Comment by lolmaus** [comment]
Nope, that's a merge and I've got no idea how to squash it.

Found some info [here](http://stackoverflow.com/questions/14043961/git-squash-history-after-merge), but both solutions are very hacky and destructive.

If you know how to do it, I add you to my fork so that you do it yourself.

**Comment by jherdman** [comment]
The following should fix ya up: `git rebase master` (assuming your master branch is in sync with upstream).

**Comment by lolmaus** [comment]
The more you know... It's done.

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "ember-cli-app-version": "^1.0.0",
     "ember-cli-content-security-policy": "0.4.0",
     "ember-cli-dependency-checker": "^1.1.0",
+    "ember-cli-es5-shim": "0.1.1",
```
Comment (jherdman) [inline package.json]: Is this merely defensive? I'm not too worried about old ass browsers.

**Comment by jherdman** [comment]
OK, so I get what's happening here, but I'm not really sure how this bug can happen. Can you give me an example scenario?

**Comment by lolmaus** [comment]
Wut? Just throw any error, and TrackJS will try to invoke unbound `handler.report()` which references `this.reporter.track` but `this.reporter` is undefined.

File: package.json (hunk 1 above)
Comment (lolmaus) [inline package.json]: It's for PhantomJS. This polyfill is in `devDependencies` and will not make it into app builds.

**Comment by lolmaus** [comment]
Thank you James! :beers:

yeeees

**Comment by lolmaus** [comment]
Will you release this?

**Comment by jherdman** [comment]
Already released! 0.2.4 is out, and 1.0 is hot on its tail (same as 0.2.4, just with the 1.0 name to indicate that I believe it's production ready).

**Comment by lolmaus** [comment]
You rock! :moyai:

**Pull request merged**